### PR TITLE
Story CORE-467 | Modify operator structure to handle multibrokers

### PR DESCRIPTION
**Pull request by LucasStarlingdePaulaSalles**
# Description

Kind: story
Section: operators

Summary: configured operator interface to dynamically configure operators using broker configuration from broker memory

## Changelog
- features:
   - configured broker memory structure do store configurations
   - configured operator manager to import configurations dynamically so that operator always work with broker configured
- fix:
    - fixed tests
    - removed unnecessary connection port from broker-specific sidecars

### CODE REVIEW

File: pkg/meta/brokers/brokers.go [hunk 1]
```diff
@@ -18,10 +15,17 @@ type ChannelBroker struct {
 }
 
 // BrokerConfiguration generic interface type
-type BrokerConfiguration interface{}
-
-// BrokerStatus generiic status type for brokers, used as parameters and returns
-type BrokerStatus string
+type BrokerConfiguration interface {
```
Comment (franciscobonand) [inline pkg/meta/brokers/brokers.go:18]: Vewy nice

File: cmd/insprd/operators/channel.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+func NewGeneralOperator(brokers brokers.Manager, memory memory.Manager) *GenOp {
+	return &GenOp{
+		brokers: brokers,
+		memory:  memory,
+		configs: make(map[string]struct {
+			config metabrokers.BrokerConfiguration
+			op     ChannelOperatorInterface
+		}),
+	}
+}
```
Comment (franciscobonand) [inline cmd/insprd/operators/channel.go:36]: Parameters can be `(scope, name string)` so it avoids unnecessary repetition

File: cmd/insprd/operators/channel.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	if obj, ok := g.configs[broker]; !reflect.DeepEqual(obj.config, config) || !ok {
+		err = g.setOperator(config)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return g.configs[channel.Spec.SelectedBroker].op, nil
+}
+
+func (g GenOp) setOperator(config metabrokers.BrokerConfiguration) error {
```
Comment (franciscobonand) [inline cmd/insprd/operators/channel.go:56]: I think this is unnecessary, as this function is only used in `getOperator` and this validation is already done there

File: cmd/insprd/operators/channel.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+				op:     operator,
+			}
+		}
+	default:
+		err = ierrors.NewError().Message("").Build()
+	}
+	return err
+}
+
+//Get executes Get method of correct operator given the desired channel's broker
```
Comment (franciscobonand) [inline cmd/insprd/operators/channel.go:77]: Parameters can be (scope, name string) so it avoids unnecessary repetition

File: cmd/insprd/operators/channel.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	return op.Create(ctx, scope, channel)
+}
+
+//Update executes Update method of correct operator given the desired channel's broker
+func (g GenOp) Update(ctx context.Context, scope string, channel *meta.Channel) error {
+	op, err := g.getOperator(scope, channel.Meta.Name)
+	if err != nil {
+		return err
+	}
+	return op.Update(ctx, scope, channel)
```
Comment (franciscobonand) [inline cmd/insprd/operators/channel.go:101]: Parameters can be (scope, name string) so it avoids unnecessary repetition

File: cmd/insprd/operators/kafka/utils.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package channels
+package kafka
```
Comment (franciscobonand) [inline cmd/insprd/operators/kafka/utils.go]: The "main" kafka lib ("github.com/confluentinc/confluent-kafka-go/kafka") is used by using `kafka.Something`, so I think it's better to rename this package to `kafkaop`or something alike

**Review by franciscobonand — CHANGES_REQUESTED** [review]

**Review by franciscobonand — APPROVED** [review]
Nice!

**Review by luiz-couto — APPROVED** [review]

**Pull request merged**